client/keys.go

```diff
@@ -91,16 +91,15 @@ func runAddCmd(cmd *cobra.Command, args []string) error {
 	)
 
 	dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun)
+	newcontext := clientCtx
 	if dryRun {
 		kr, err = keyring.New(sdk.KeyringServiceName(), keyring.BackendMemory, clientCtx.KeyringDir, buf, hd.EthSecp256k1Option())
-	} else {
-		backend, _ := cmd.Flags().GetString(flags.FlagKeyringBackend)
-		kr, err = keyring.New(sdk.KeyringServiceName(), backend, clientCtx.KeyringDir, buf, hd.EthSecp256k1Option())
+		newcontext = clientCtx.WithKeyring(kr)
 	}
 
 	if err != nil {
 		return err
 	}
 
-	return clientkeys.RunAddCmd(clientCtx.WithKeyring(kr), cmd, args, buf)
+	return clientkeys.RunAddCmd(newcontext, cmd, args, buf)
```

Comment:
use an updated clientCtx

Reply:
it's for just using original keyring backend which is from client.toml
(possible modified by flag --keyring-backend)

without this code, user initial setting is not used